src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -6,55 +6,99 @@
 
 public class Range<T> implements Set<T> {
 
+    ArrayList<T> arrayList = new ArrayList<>();
+
     public int size() {
-        return 0;
+        return arrayList.size();
     }
 
     public boolean isEmpty() {
-        return false;
+
+        return arrayList.isEmpty();
     }
 
     public boolean contains(Object o) {
-        return false;
+        return arrayList.contains(o);
     }
 
     public Iterator<T> iterator() {
-        return null;
+        return arrayList.iterator();
     }
 
     public Object[] toArray() {
-        return new Object[0];
+
+        return arrayList.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+
+        return (T1[]) arrayList.toArray();
     }
 
     public boolean add(T t) {
-        return false;
+        return  arrayList.add(t);
```

Comment:
You have to check if the element is unique.

Reply:
public boolean add(T t) {
        if (arrayList.contains(t)) return false;
        return  arrayList.add(t);
    }

Reply:
Ok.

Reply:
already updated